### PR TITLE
Log errors by default

### PR DESCRIPTION
**Pull request by tamird**
Remove imperative `clap::Error::exit`.

### CODE REVIEW

**Comment by tamird** [comment]
@vadorovsky looks like eff4440f73aa380eba658a5b5a93f5fe4e298a2e accidentally lost logging if an explicit level was not passed to the CLI.

**Review by vadorovsky — COMMENTED** [review]
Awesome, happy to see the custom `error` function to be gone and replaced with anyhow! So far it looks good, I'm just going to try it a bit before approving.

File: src/bin/bpf-linker.rs [hunk 1]
```diff
@@ -192,57 +216,40 @@ fn main() {
...
                 let subscriber = subscriber_registry.with(tracing_layer(io::stderr));
-                (Box::new(subscriber), None)
+                (either::Right(subscriber), None)
             }
         };
-        if let Err(e) = tracing::subscriber::set_global_default(subscriber) {
-            error(&e.to_string(), clap::error::ErrorKind::Format);
-        }
-    }
+        either::for_both!(subscriber, s => tracing::subscriber::set_global_default(s))?;
```
Comment (alessandrod) [inline src/bin/bpf-linker.rs:243]: unless it's already in the dep tree, it doesn't seem worth adding a new dep just for this?

**Review by alessandrod — COMMENTED** [review]
lgtm but let's let @vadorovsky try what he wants to try.

As per comment I wouldn't add a new dep tho.

File: src/bin/bpf-linker.rs (hunk 1 above)
Comment (tamird) [reply src/bin/bpf-linker.rs:243]: Take a look at Cargo.lock; it's already in there.

Comment (alessandrod) [reply src/bin/bpf-linker.rs:243]: It's a dev dependency right?

Comment (tamird) [reply src/bin/bpf-linker.rs]: Good point, removed.

**Pull request merged**